File: backend/components/metiers/threed/templates/helpers/utils/templates_utils.js
/**
 * templates_utils.js – Logique métier ultra avancée helpers/utils (JS)
 * CDC Dihya : RGPD, accessibilité, audit, CI/CD, edge cases, hooks, robustesse, conformité, multi-formats
 */
const crypto = require('crypto');

function generateTemplateId(prefix = 'tpl') {
  const date = new Date().toISOString().replace(/[-:.TZ]/g, '').slice(0, 17);
  const rand = crypto.randomBytes(3).toString('hex');
  return `${prefix}_${date}_${rand}`;
}

function cleanTemplateContent(content) {
  if (!content) return '';
  let cleaned = content.replace(/[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}/g, '[anonymisé]');
  cleaned = cleaned.replace(/Nom: [^\n]+/g, 'Nom: [anonymisé]');
  return cleaned;
}

function checkAccessibility(content) {
  return /aria-|alt=|role=/.test(content || '');
}

function auditTemplate(content) {
  return {
    length: (content || '').length,
    hasAccessibility: checkAccessibility(content || ''),
    isRGPD: !(content || '').includes('@'),
    timestamp: new Date().toISOString()
  };
}

module.exports = {
  generateTemplateId,
  cleanTemplateContent,
  checkAccessibility,
  auditTemplate
};

File: backend/components/metiers/threed/templates/helpers/utils/templates_utils.py
"""
templates_utils.py – Logique métier ultra avancée helpers/utils (Python)
CDC Dihya : RGPD, accessibilité, audit, CI/CD, edge cases, hooks, robustesse, conformité, multi-formats
"""
import re
from datetime import datetime
import random
import string

def generate_template_id(prefix='tpl'):
    date = datetime.utcnow().strftime('%Y%m%d%H%M%S%f')
    rand = ''.join(random.choices(string.ascii_lowercase + string.digits, k=6))
    return f"{prefix}_{date}_{rand}"

def clean_template_content(content):
    if not content:
        return ''
    content = re.sub(r"[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}", '[anonymisé]', content)
    content = re.sub(r"Nom: [^\n]+", 'Nom: [anonymisé]', content)
    return content

def check_accessibility(content):
    return bool(re.search(r"aria-|alt=|role=", content or ''))

def audit_template(content):
    return {
        'length': len(content or ''),
        'hasAccessibility': check_accessibility(content or ''),
        'isRGPD': '@' not in (content or ''),
        'timestamp': datetime.utcnow().isoformat()
    }
